source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -63,10 +78,12 @@ Http::FilterHeadersStatus CacheFilter::decodeHeaders(Http::RequestHeaderMap& hea
   }
   ASSERT(decoder_callbacks_);
 
+  initConfig();
```

Comment:
No need to initConfig in encodeHeaders because you can't all of
1. get here
2. still need a cache
3. not have gone through the initConfig in decodeHeaders

(Some filters do need to do it in both paths because an early-out in decodeHeaders can mean they still need the config for the reply, but the only early-outs in encodeHeaders here involve either already having the cache or not needing it.)

Reply:
It is not only about cache but also other config parameters. If situation is possible: 'init_config has not been called in decode_headers' then I would prefer to leave this config initialization here.